test/surface_test.py

```diff
@@ -757,41 +814,83 @@ def test_blit__SRCALPHA_opaque_source(self):
 
-        s = pygame.Surface(size, pygame.SRCALPHA, 32)
-        s2 = s.copy()
+        for x in range(area.width):
+            for y in range(area.height):
+                (r, g, b, a) = reference_color = reference_surface.get_at((x,y))
+                expected_color = (r, g, b, (a+(a*((256-a)//256))))
+                self.assertEqual(reference_color, expected_color)
 
-        s.fill((32, 32, 32, 111))
-        s2.fill((32, 32, 32, 31))
+        self.assertEqual(reference_surface.get_rect(), test_surface.get_rect())
 
-        s.blit(s2, (0, 0))
 
-        # TODO:
-        # what is the correct behaviour ?? should it blend? what algorithm?
+    def test_blit__SRCALPHA_to_SRCALPHA_non_zero(self):
+        """Tests blitting a nonzero alpha surface to another nonzero alpha surface
+         both straight alpha compositing method. Test is fuzzy (+/- 1/256) to account for
+         different implementations in SDL1 and SDL2.
+        """
 
-        self.assertEqual(s.get_at((0, 0)), (32, 32, 32, 31))
+        size = (32, 32)
+
+        def check_color_diff(color1, color2):
+            """Returns True if two colors are within (1, 1, 1, 1) of each other."""
+            for val in (color1 - color2):
+                if abs(val) > 1:
+                    return False
+            return True
+
+        def high_a_onto_low(high, low):
+            """Tests straight alpha case. Source is low alpha, destination is high alpha"""
+            high_alpha_surface = pygame.Surface(size, pygame.SRCALPHA, 32)
+            low_alpha_surface = high_alpha_surface.copy()
+            high_alpha_color = Color((high, high, low, high)) # Injecting some RGB variance. Any numbers can be chosen.
+            low_alpha_color = Color((high, low, low, low))
+            high_alpha_surface.fill(high_alpha_color)
+            low_alpha_surface.fill(low_alpha_color)
+            
+            high_alpha_surface.blit(low_alpha_surface, (0, 0))
+            
+            expected_color = (low_alpha_color) + Color(tuple(((x*(255-low_alpha_color.a))//255) for x in high_alpha_color))
+            self.assertTrue(check_color_diff(high_alpha_surface.get_at((0, 0)), expected_color))
+        
+        def low_a_onto_high(high, low):
+            """Tests straight alpha case. Source is high alpha, destination is low alpha"""
+            high_alpha_surface = pygame.Surface(size, pygame.SRCALPHA, 32)
+            low_alpha_surface = high_alpha_surface.copy()
+            high_alpha_color = Color((high, high, low, high)) # Injecting some RGB variance. Any numbers can be chosen.
+            low_alpha_color = Color((high, low, low, low))
+            high_alpha_surface.fill(high_alpha_color)
+            low_alpha_surface.fill(low_alpha_color)
+    
+            low_alpha_surface.blit(high_alpha_surface, (0, 0))
+            
+            expected_color = (high_alpha_color) + Color(tuple(((x*(255-high_alpha_color.a))//255) for x in low_alpha_color))
```

Comment:
It's in general a good idea to avoid using other modules when testing as much as possible. But here `high_alpha_color.a` is more readable than `high_alpha_color[3]`.